docs/migrate-from-cluster.md

```diff
@@ -0,0 +1,226 @@
+      sudo mv /etc/docker/daemon.json /etc/docker/daemon.json.backup
+      ```
+
+    * Masters only
+
+      Create `/var/log/kube-audit`
+
+      ```ShellSession
+      sudo mkdir var/log/kube-audit
+      ```
+
+      Change ETCD version to `quay.io/coreos/etcd:v3.4.13`
+
+      **NOTE**: This will restart etcd and if you only have one master it will stop the api server until it's restarted.
+
+      ```ShellSession
+      sudo vim /etc/kubernetes/manifest/etcd.yaml
+      ```
+
+1. Create inventory for the old cluster
+
+    Add the old machines to the new kubespray config path.
+
+    Remember to add
+
+    ```ini
+    [kube-master:vars]
+    supplementary_addresses_in_ssl_keys = [ "<LB public IP>", "<LB private IP>" ]
+    ```
+
+1. Modify default group_vars to match old setup
+
+    Add the following to the `k8s-cluster` group
+
+    ```yaml
+    etcd_kubeadm_enabled: true
+    kube_cert_dir: "/etc/kubernetes/pki"
+    kube_service_addresses: "10.96.0.0/12"
+    kube_pods_subnet: "192.168.0.0/16"
+
+    audit_log_hostpath: "/var/log/kube-audit"
+    audit_log_path: "/var/log/kube-audit/kube-apiserver.log"
+    audit_policy_file: "/etc/kubernetes/conf/audit-policy.yaml"
+    ```
+
+1. Run kubespray on the old machines
+
+    ```ShellSession
+    export CK8S_CONFIG_PATH=/path/to/new/config
+
+    bin/ck8s-kubespray run-playbook ${ENVIRONMENT_NAME} cluster.yml
+    ```
+
+    Verify that all kubernetes components are running and healthy after this step.
+
+1. Run terraform to set up new machines
+
+1. Add new workers
+
+    ```ShellSession
+    export CK8S_CONFIG_PATH=/path/to/new/config
+
+    bin/ck8s-kubespray run-playbook ${ENVIRONMENT_NAME} fetch.yml
+    bin/ck8s-kubespray run-playbook ${ENVIRONMENT_NAME} scale.yml --limit=new-worker-X # Add one worker at a time
+    ```
+
+    Between each new worker, verify that they get added to kubernetes and get ready
+
+1. Add temporary security role between the old and the new machines
+
+    Manually create new rules bewteen the security groups `${ENVIRONMENT_NAME}-cluster` and `${ENVIRONMENT_NAME}-k8s` to allow all traffic
+
+1. Add new masters
+
+    ```ShellSession
+    export CK8S_CONFIG_PATH=/path/to/new/config
+
+    bin/ck8s-kubespray run-playbook ${ENVIRONMENT_NAME} cluster.yml --limit=etcd,kube-master -e ignore_assert_errors=yes -e etcd_retries=10
+    bin/ck8s-kubespray run-playbook ${ENVIRONMENT_NAME} upgrade-cluster.yml --limit=etcd,kube-master -e ignore_assert_errors=yes -e etcd_retries=10
+    ```
+
+    Verify that all the new masters are added and that all kubernetes components are running successfully.
+
+1. Remove old workers
+
+    ```ShellSession
+    export CK8S_CONFIG_PATH=/path/to/new/config
+
+    bin/ck8s-kubespray run-playbook ${ENVIRONMENT_NAME} remove-node.yml -e node=old-worker-X # Remove one worker at a time
+    ```
+
+1. Remove old master
+
+    **NOTE**: Make sure you are not removing more than half the masters.
+    That will make you loose quorum.
```

Comment:
I think it would be good to also mention the [limitation about removing the first master](https://kubespray.io/#/docs/nodes?id=limitation-removal-of-first-kube-master-and-etcd-master)

Reply:
Yes! I was going to add that, but it must've slipped my mind.